packages/apps/graph-client/src/utils/hooks/use-parsed-blocks.ts

```diff
@@ -0,0 +1,50 @@
+import type { Block } from '../../__generated__/sdk';
+
+import { useCallback, useState } from 'react';
+
+interface IUseParseBlocksReturn {
+  allBlocks: Record<number, Block[]>;
+  addBlocks: (blocks: Block[]) => void;
+}
+
+export function useParsedBlocks(): IUseParseBlocksReturn {
+  const [allBlocks, setAllBlocks] = useState<Record<number, Block[]>>({});
```

Comment:
You're using the `Block` type here. This is a nice and easy way to define the interface for `useParsedBlocks` but it's confusing and breaks typings.

You're essentially saying: `addBlocks` can accept only blocks that satisfy the `Block` type can be passed, but that's not what you want.

Therefore, it's best to make an interface that is _just_ what you need for this code to work.

Based on the implementation, you probably want this interface:

```ts
interface IBlock {
  height: number;
  chainid: number;
}
```

Reply:
makes sense! It has been updated to what you suggested